viewing_party/party.py

```diff
@@ -1,23 +1,180 @@
 # ------------- WAVE 1 --------------------
-
+#adding a comment in the code to commit :)
 def create_movie(title, genre, rating):
-    pass
+    
+    user_data = {
+        "watched": [],
+        "watchlist": []
+        }
+
+    new_movie = {
+        "title": title,
+        "genre": genre,
+        "rating": rating
+    }
+    if None in new_movie.values():
+        return None
+
+    return new_movie
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+    
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    watchlist_length = len(user_data['watchlist'])
+
+    """compare the title argument to value of title in
+    each movie dictionary in the user's watchlist
+    if the titles are the same then append the movie dictionary
+    to the user's watched list and delete the dictionary
+    from the user's watchlist"""
+
+    for i in range(watchlist_length):
+        if user_data['watchlist'][i]['title'] == title:  
+            user_data['watched'].append(user_data['watchlist'][i])
+            del user_data['watchlist'][i]
+    
+    return user_data
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    avg_rating = 0.0
+    sum_rating = 0
+    watched_length = len(user_data['watched'])
+
+    #exit function if user's watched list is empty
+    if watched_length < 1:
+        return avg_rating
+    
+    for movie in user_data['watched']:
+        sum_rating += movie["rating"]
 
+    avg_rating = sum_rating/watched_length
+    
+    return avg_rating
 
+def get_most_watched_genre(user_data):
+    popular_genre = None

```

Comment:
I would consider grouping `popular_genre` and `popular_count` together because they are related OR combining that information as a dictionary!